client/extensions/woocommerce/app/settings/email/email-settings/index.js

```diff
@@ -85,75 +102,113 @@ const customerNotifications = [
 ];
 
 class Settings extends React.Component {
-
-	fetchSettings = ( props ) => {
+	fetchSettings = props => {
 		const { siteId, fetchSettings } = props;
 		siteId && fetchSettings( siteId );
-	}
+	};
 
 	componentDidMount = () => {
 		this.fetchSettings( this.props );
 	};
 
-	componentWillReceiveProps = newProps => {
-		if ( newProps.siteId !== this.props.siteId ) {
-			this.fetchSettings( newProps );
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.siteId !== this.props.siteId ) {
+			this.fetchSettings( nextProps );
 		}
-	};
 
-	notificationsToggle = () => {
-		undefined;
-	}
+		// Save settings request
+		if ( ! this.props.saveSettingsRequest && nextProps.saveSettingsRequest ) {
+			const settingsClean = omit( nextProps.settings, [
+				'save',
+				'isSaving',
+				'error',
+				'invalidValue',
+			] );
+			const areSettingsValid = validateSettings( settingsClean );
+			if ( nextProps.loaded ) {
+				if ( ! areSettingsValid ) {
+					nextProps.emailSettingsInvalidValue(
+						nextProps.siteId,
+						'Email Settings are invalid: please correct.'
+					);
+					nextProps.errorNotice( translate( 'Please correct your Email settings and try again.' ) );
+				} else {
+					nextProps.submit( nextProps.siteId, nextProps.settings );
+				}
+			} else {
+				nextProps.emailSettingsInvalidValue(
+					nextProps.siteId,
+					'Email Settings not loaded: please wait.'
+				);
+			}
+		}
 
-	recipientsChange = () => {
-		undefined;
+		// Settings save request finished.
+		if ( ! nextProps.isSaving && this.props.isSaving ) {
+			if ( nextProps.submitError ) {
+				nextProps.errorNotice(
+					translate( 'There was a problem saving the email settings. Please try again.' )
+				);
+			} else {
+				nextProps.successNotice( translate( 'Email settings saved.' ), { duration: 4000 } );
```

Comment:
Hi! I've found a possible matching string that has already been translated **95** times:
`translate( 'Settings saved.' )` *ES Score: 7*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/email-settings-ui-save)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).